### PR TITLE
Remove deprecated IBMQ dependent functionality

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit removes all the functionality in Qiskit that depended on the legacy qiskit-ibmq-provider package. This process was started in #10525 which removed the qiskit.IBMQ alias object but as was discovered in that PR there were many other places in qiskit that were relying on the qiskit-ibmq-provider package. This commit removes all of these places, which primarily include qiskit.test.decorators's online_test decorator, qiskit.tools.jupyter, qiskit.tools.monitor, qiskit.test.ibmq_mock, and qiskit.test.mock (which just referenced IBMQ and didn't actually use it). With these removals we're also no longer optionally dependent on ipywidgets, jupyter, or ipython and those are removed from the optional and development requirements lists.

### Details and comments

### CODE REVIEW

File: qiskit/__init__.py [hunk 1]
```diff
@@ -74,8 +74,7 @@
 
 _config = _user_config.get_config()
 
-# Moved to after IBMQ and Aer imports due to import issues
-# with other modules that check for IBMQ (tools)
+# Moved to after  Aer imports due to import issues
```
Comment (jakelishman) [inline qiskit/__init__.py]: Can probably delete the comment since it's not accurate even modified.

File: qiskit/providers/__init__.py [hunk 2]
```diff
@@ -462,7 +462,7 @@ def get_translation_stage_plugin(self):
 handles submitting the circuits to the backend to be executed and returning a
 :class:`~qiskit.providers.Job` object. Depending on the type of backend this
 typically involves serializing the circuit object into the API format used by a
-backend. For example, on IBMQ backends from the ``qiskit-ibmq-provider``
+backend. For example, on IBM backends from the ``qiskit-ibm-provider``
 package this involves converting from a quantum circuit and options into a
 `qobj <https://arxiv.org/abs/1809.03452>`__ JSON payload and submitting
 that to the IBM Quantum API. Since every backend interface is different (and
```
Comment (jakelishman) [inline qiskit/providers/__init__.py]: I'm amazed this is still accurate.

File: qiskit/providers/fake_provider/__init__.py [hunk 3]
```diff
@@ -68,19 +68,6 @@
...
-        from qiskit.providers.aer import AerSimulator
-
-        # get a real backend from a real provider
-        provider = IBMQ.load_account()
-        backend = provider.get_backend('ibmq_manila')
-
-        # generate a simulator that mimics the real quantum system with the latest calibration results
-        backend_sim = AerSimulator.from_backend(backend)
-
-
```
Comment (jakelishman) [inline qiskit/providers/fake_provider/__init__.py]: The paragraph above says "please follow steps below to ...", so its text probably needs changing to something as well.

**Review by jakelishman — COMMENTED** [review]
This has a large amount of overlap with #11445, but that's no problem.  Mostly this looks fine to me, just very minor comments.

File: qiskit/providers/__init__.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/providers/__init__.py]: Oh actually it isn't I misread it, the qobj part isn't valid anymore the qiskit-ibm-provider is only submitting circuits via qpy over the wire. The rest of that paper about configuration, properties, and pulse defaults are valid though and that's how the backends are described to the provider. I'll update this.

File: qiskit/providers/fake_provider/__init__.py (hunk 3 above)
Comment (mtreinish) [reply qiskit/providers/fake_provider/__init__.py]: Done in: https://github.com/Qiskit/qiskit/pull/11513/commits/f4e26f9afa9bd8bc195a4914017809241110d871

File: qiskit/providers/__init__.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/providers/__init__.py]: Fixed in: https://github.com/Qiskit/qiskit/pull/11513/commits/f4e26f9afa9bd8bc195a4914017809241110d871

File: qiskit/__init__.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/__init__.py]: Fixed in: https://github.com/Qiskit/qiskit/pull/11513/commits/f4e26f9afa9bd8bc195a4914017809241110d871

**Review by jakelishman — APPROVED** [review]
This looks like as clean a removal as we'll get of these.  Good to tidy these things up before 1.0.

**Pull request merged**